### PR TITLE
Wait for velero deployment to be ready

### LINKED ISSUE #3861: velero install does not wait for velero to be healthy and does not report errors
**What steps did you take and what happened:**
Did a velero install, but there is an error pulling the image.

```
$ kubectl get all -n velero
NAME                          READY   STATUS              RESTARTS   AGE
pod/restic-58jtz              0/1     ContainerCreating   0          11s
pod/velero-5b9775799f-pt7w4   0/1     Init:ErrImagePull   0          11s

NAME                    DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
daemonset.apps/restic   1         1         0       1            0           <none>          11s

NAME                     READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/velero   0/1     1            0           11s

NAME                                DESIRED   CURRENT   READY   AGE
replicaset.apps/velero-5b9775799f   1         1         0       11s
```

Velero install returned with no errors.  

This can easily be reproduced by giving a bad image name.

**What did you expect to happen:**

Error that the installation is not coming up

**The output of the following commands will help us better understand what's going on**:
(Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)

- `kubectl logs deployment/velero -n velero`

`kubectl logs deployment/velero -n velero
Error from server (BadRequest): container "velero" in pod "velero-5b9775799f-pt7w4" is waiting to start: PodInitializing`

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

### PR DESCRIPTION
**Pull request by Divya063**
Signed-off-by: Divya Rani <drani@vmware.com>

Thank you for contributing to Velero!

# Please add a summary of your change

Running `velero install` does not wait for velero to be healthy and does not report errors. The changes fix this issue. 

# Does your change fix a particular issue?
Yes

Fixes #3861 

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required`.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Review by jenting — COMMENTED** [review]
Could we wait for restic DaemonSet if `--use-restic` be passed?
Besides that, the changelog is required. Thank you.

**Comment by Divya063** [comment]
@jenting Sure, I see a [wait flag](https://github.com/vmware-tanzu/velero/blob/d871370c1e7e265acac418d70e96fa5182c9fb7f/pkg/cmd/cli/install/install.go#L281) for that purpose ( for both velero deployment and restic Daemonset), I was wondering shouldn't it be a default behaviour?

**Comment by jenting** [comment]
Or perhaps the original usage did not pass the `--wait` flag?

**Comment by Divya063** [comment]
@jenting  I was thinking if we could remove the wait flag itself, because anyway we would be waiting for both velero deployment and restic Daemonset (after I make the change)?

**Comment by jenting** [comment]
Please do not break backward compatible.

**Comment by Divya063** [comment]
@jenting I have made the changes. Thanks!

File: pkg/cmd/cli/install/install.go [hunk 1]
```diff
@@ -278,6 +278,13 @@ func (o *InstallOptions) Run(c *cobra.Command, f client.Factory) error {
 		return errors.Wrap(err, errorMsg)
 	}
 
+	if o.UseRestic {
+		fmt.Println("Waiting for Velero restic daemonset to be ready.")
+		if _, err = install.DaemonSetIsReady(factory, o.Namespace); err != nil {
+			return errors.Wrap(err, errorMsg)
+		}
+	}
+
```
Comment (jenting) [inline pkg/cmd/cli/install/install.go:287]: Isn't it duplicate? https://github.com/vmware-tanzu/velero/blob/002bc56ad360c6b3604bc7c9508b7252ebe6f53e/pkg/cmd/cli/install/install.go#L294-L299

**Review by jenting — CHANGES_REQUESTED** [review]
I'd suggest checking the original issue's bug _or_ request, then find an appropriate way to fix it. Thank you.

**Pull request closed without merging**